node/node.go

```diff
@@ -201,6 +201,16 @@ func GetCommand() *cobra.Command {
 			// os.Interrupt for all systems, especially windows, syscall.SIGTERM is mainly for docker.
 			ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
 			defer cancel()
+
+			// listen for \q in stdin to gracefully shutdown the node
+			// see https://github.com/spacemeshos/go-spacemesh/issues/5321
+			go func() {
+				var input string
+				if _, err := fmt.Scanln(&input); err == nil && input == "\\q" {
+					cancel()
+				}
+			}()
```

Comment:
If we want to go the route of sending a signal via `stdin` to shut down a node we should make it `CTRL + D` if we can:

```suggestion
			app.eg.Go(func() error {
				for {
					select {
					case <-ctx.Done():
						return nil
					default:
						buf := make([]byte, 10)
						_, err := os.Stdin.Read(buf)
						if err != nil {
							cancel()
							return nil
						}
					}
				}
			})
```

This keeps track of the spawned go routine (and ensures it exits before the node terminates) and causes any input on `os.Stdin` to be ignored/discarded.

Triggering the graceful shutdown can be done by closing `stdin` of the process (I assume calling `end` on `child.stdin.end()` in electron?). Additionally when running the node in a terminal `CTRL + D` will also trigger a shutdown  🙂

Reply:
CTRL+C already works well in console.
But the problem is that when we run go-spacemesh as a child process — it is not attached to the console, and that's why we can't use `GeneralConsoleCtrlEvent` or something similar.

Closing stdin instead of printing `\q` may work. I need to check it out.

Reply:
The proposed code and solution (closing stdin) works well on MacOS.

Reply:
Any chance you can also try it on Windows? 🙂

Reply:
Yep, just did it. Working well! 👍